src/components/LocaleSwitcher.tsx

```diff
@@ -0,0 +1,65 @@
+'use client';
+import 'flag-icons';
+import { MouseEvent, useState } from 'react';
+import { useRouter } from 'next/router';
+import { Locale, i18n } from 'i18n.config';
+import { usePathname } from 'next/navigation';
+
+export default function LocaleSwitcher(props: {
+  lang: Locale;
+  mobile: boolean;
+}) {
+  const router = useRouter();
+  const pathname = usePathname();
+
+  const defaultLocale = i18n.defaultLocale;
+  const [animation, setAnimation] = useState('');
+  const [locale, setLocale] = useState(props.lang);
+
+  const toggle = (e: MouseEvent<HTMLDivElement>) => {
+    const container = e.target as HTMLDivElement;
+    const changeLocale = i18n.locales.filter(
+      locale => locale !== props.lang
+    )[0];
+
+    setLocale(changeLocale);
+    setAnimation(props.lang === defaultLocale ? 'slide-out' : 'slide-in');
+
+    container.onanimationend = e => {
+      const regex = /it|en/;
+
+      if (regex.test(pathname)) {
+        // not works: onclick runs twice ######
```

Comment:
Hi @salvatorericcardi! If for some reason you're going to change the code again, I'd suggest you to remove useless 'console.log' comments here and there.